food_ordering_app/app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,82 @@
+class CartsController < ApplicationController
+  before_action :set_cart_item, only: [:update, :destroy]
+
+  def index
+    @cart_items = current_user.cart_items.includes(:product)
+    @cart_total = @cart_items.sum { |item| item.total_price }
+  end
+
+  def create
+    product = Product.find(params[:product_id])
+    quantity = (params[:quantity] || 1).to_i
+    
+    if quantity <= 0
+      redirect_to root_path, alert: "Invalid quantity."
+      return
+    end
+    
+    existing_item = current_user.cart_items.find_by(product: product)
+    
+    if existing_item
+      new_quantity = existing_item.quantity + quantity
+      if existing_item.update(quantity: new_quantity)
+        redirect_to carts_path, notice: "Item quantity updated in cart successfully."
+      else
+        redirect_to root_path, alert: "Failed to update cart item."
+      end
+    else
+      @cart_item = current_user.cart_items.build(product: product, quantity: quantity)
+      if @cart_item.save
+        redirect_to carts_path, notice: "Item added to cart successfully."
+      else
+        redirect_to root_path, alert: "Failed to add item to cart."
+      end
+    end
+  rescue ActiveRecord::RecordNotFound
+    redirect_to root_path, alert: "Product not found."
+  end
+
+  def destroy
```

Comment:
You could move the destroy method after update to follow the default rails methods order `index show new edit create update destroy`